examples/v2/matterbridge/chat2bridge.nim

```diff
@@ -243,14 +243,20 @@ when isMainModule:
     (nodev2ExtIp, nodev2ExtPort, _) = setupNat(conf.nat, clientId,
                                                Port(uint16(conf.libp2pTcpPort) + conf.portsShift),
                                                Port(uint16(conf.udpPort) + conf.portsShift))
+    ## The following heuristic assumes that, in absence of manual
+    ## config, the external port is the same as the bind port.
+    extPort = if nodev2ExtIp.isSome() and nodev2ExtPort.isNone():
```

Comment:
In a roundabout way, yes. They assign the external port to the bind port [as part of NAT address setup](https://github.com/status-im/nim-eth/blob/2dfd352fd0635171f8cb9a86db94b352d63012e5/eth/net/nat.nim#L298-L299).
Since we use a different utility, we do the port assignment after the NAT setup (also [for wakunode2](https://github.com/status-im/nim-waku/blob/master/waku/v2/node/wakunode2.nim#L820-L826)).

It might be worth changing our logic to match nimbus2 exactly, though for now I've tried to remain consistent with what we have.